L/libdatachannel/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libdatachannel"
+version = v"0.20.2"
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/paullouisageneau/libdatachannel.git", "9cbe6a2a1f21cde901bca9571581a96c6cda03cf"),
+    ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.14.sdk.tar.xz", 
+               "0f03869f72df8705b832910517b47dd5b79eb4e160512602f593ed243b28715f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+if [[ ${target} == x86_64*mingw* ]]; then
+    export OPENSSL_ROOT_DIR="${prefix}/lib64"
+fi
+
+if [[ "${bb_full_target}" == x86_64-apple-darwin* ]]; then
+    # LLVM 15+ requires macOS SDK 10.14.
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    export MACOSX_DEPLOYMENT_TARGET=10.14
+    popd
+fi
+
+cd $WORKSPACE/srcdir/libdatachannel
+git submodule update --init --recursive --depth 1
+cmake -B build \
+    -DUSE_GNUTLS=0 \
+    -DUSE_NICE=0 \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release
+cmake --build build --parallel ${nproc}
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libdatachannel", :libdatachannel)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="OpenSSL_jll", uuid="458c3c95-2e84-50aa-8efc-19380b2a3a95"); compat="3.0.8")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"13.2.0")
```

Comment:
GCC version 13.2.0 is very restrictive, which is what causes this error:
```
ERROR: could not load library "/cache/build/yggy-amdci7-15/julialang/yggdrasil/L/libdatachannel/build/x86_64-linux-gnu-cxx11/5iGi6pG4/x86_64-linux-gnu-libgfortran5-cxx11/destdir/lib/libdatachannel.so.0.20.2"
/cache/julia-buildkite-plugin/julia_installs/bin/linux/x64/1.7/julia-1.7-latest-linux-x86_64/bin/../lib/julia/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /cache/build/yggy-amdci7-15/julialang/yggdrasil/L/libdatachannel/build/x86_64-linux-gnu-cxx11/5iGi6pG4/x86_64-linux-gnu-libgfortran5-cxx11/destdir/lib/libdatachannel.so.0.20.2)
```

Here it is trying to use the C++ runtime library version 3.4.32, but that is the most recent one and so will only work on the most recent systems. The general rule is that libraries built in Yggdrasil should use the oldest possible GCC version to build with to ensure they can be used on the largest number of systems without compatibility issues.

I would suggest starting with GCC 7 and increasing the version number until you find a compiler version that works without error (we have a mix of GCC 7, 8 and 9 in use for C++17 programs, depending on the exact C++17 features they use).

Reply:
@imciner2 Wait, is the `CompilerSupportLibraries_jll` still needed if this works? :thinking:

Reply:
`CompilerSupportLibraries_jll` is always needed if the library links to `libgcc_s`, regardless of the version.